doc: add release notes for 1.6.0

b-gent opened the pull request:
This 1.6.0 release file includes changes from the other currently open PRs that change the release_notes_latest file.

b-gent commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,473 @@
+Highlights
+**********
+
+* Added production support for nRF5340 for Bluetooth LE in single protocol configuration and in multiprotocol operation with Thread for Bluetooth Peripheral role.
```
@jdstapes  @shantha-14  please check

thomasstenersen replied:
@emidttun FYI.

shantha-14 replied:
@thomasstenersen / @jostn should the second part of sentence be removed. We know the root cause now.

thomasstenersen replied:
No, keep as is. The issue is more likely to present itself in the multiprotocol configuration.

b-gent commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,473 @@
+* Integrated Apple HomeKit ADK v5.3 and added development support for nRF52832 and nRF52833 for Bluetooth LE HomeKit accessories.
+* Zigbee Network Co-Processor (NCP) design is now supported for production for nRF52833 and nRF52840.
+* The application layer part of ZBOSS Zigbee stack is available in source code now.
+* New sample :ref:`modem_shell_application` can be used to test various connectivity features.
```
This is a suggestion from @bjorn-tore-taraldsen

b-gent replied:
accepted

b-gent commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,473 @@
+
+This version of the |NCS| has been tested with the following modem firmware for cellular IoT applications:
+
+* mfw_nrf9160_1.3.0
```
@hani please confirm

nordic-hani replied:
Remove reference to specific modem version.
I suggest a line like this:
See :ref:`Modem firmware compatibility matrix` for an overview of which modem firmware versions have been tested with this version of |NCS|.

Add to nrf/links.txt:
.. _`Modem firmware compatibility matrix`: https://infocenter.nordicsemi.com/topic/comp_matrix_nrf9160/COMP/nrf9160/nrf9160_modem_fw.html

b-gent replied:
changed

b-gent commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+Bluetooth LE
+------------
+
+* Added:
```
From #4956  @KAGA164  please check

b-gent commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+nRF IEEE 802.15.4 radio driver
+++++++++++++++++++++++++++++++
+
+* Added production support for nRF5340 in multiprotocol configuration (IEEE 802.15.4 and Bluetooth Peripheral Role).
```
FRom #4963  @czeslawmakarski  please check

b-gent commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+Trusted Firmware-M
+------------------
+
+* Added support for hardware-accelerated cryptography in TF-M using the Nordic Security module (nrf_security).
```
From #4952  @oyvindronningstad

b-gent commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+
+* Migrated some of the application modules of nRF Desktop application to :ref:`lib_caf` for reuse by other applications.
+
+Hardware flash write protection
```
From #4952 @oyvindronningstad

b-gent commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+* PDN socket API is deprecated.
+  The functionality has been replaced by the :ref:`pdn_readme` library.
+
+Crypto
```
From #4952 @oyvindronningstad

b-gent commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+  * Reworked the :ref:`NCP sample <zigbee_ncp_sample>` to work with the simple gateway application.
+  * Moved the `NCP Host documentation`_ from the `ZBOSS NCP Host`_ package to the same location as the `external ZBOSS development guide and API documentation`_.
+
+nRF Desktop
```
From #4904  @pdunaj

b-gent commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+  * New library for managing and using hardware unique keys (HUKs), building on the APIs in nrf_cc3xx_platform.
+    HUKs are secret keys that are kept hidden from the application code, but which can be used by the application for deriving keys for other purposes, such as encrypting data for storing.
+
+Edge Impulse
```
From #4904 @pdunaj

b-gent commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+* Added the :ref:`nrf_machine_learning_app` application that integrates the Edge Impulse wrapper library with sensor sampling.
+* Sample reference to a public pre-trained model.
+
+Pelion
```
From #4904 @pdunaj

b-gent commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+  For more information, see :ref:`ug_pelion`.
+* Added the :ref:`pelion_client` application that showcases the usage of Pelion Device Management library.
+
+Common Application Framework (CAF)
```
From #4904 @pdunaj

b-gent commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+
+The following changes are relevant for all device families.
+
+Bluetooth mesh
```
From #4856  @Balaklaka  @trond-snekvik

trond-snekvik replied:
@b-gent I believe Bluetooth Mesh should go in the nRF5 section, between Bluetooth LE and Matter. It does not run on nRF91.

b-gent replied:
Moved, thanks

oyvindronningstad commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+
+  * This configuration requires 32 kB of flash and provides random number generation, SHA-256, and the platform memory read service, which is analogous to the feature set of :ref:`secure_partition_manager`.
+  * The configuration is showcased in the :ref:`tfm_hello_world` sample.
+
```
```suggestion
* The TF-M protected storage service is now using non-static labels when deriving encryption keys.
  The derivation labels are generated by combining the client id of the requesting service and the UID of the resource.
* It is now possible to run PSA tests (psa-arch-tests) and TF-M regression tests in NCS.
  The tests can be found as samples in zephyr/samples/tfm_integration.

```

oyvindronningstad replied:
@Vge0rge @ioannisg  ^

Vge0rge replied:
That seems great, and maybe we can add a comment regarding the nRF9160 configuration:
The TF-M protected storage on nRF9160 is now configured to use AES CCM to perform authenticated encryption/decryption.

b-gent replied:
added

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-4966
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

trond-snekvik commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+************
+
+Known issues are only tracked for the latest official release.
+See `known issues for nRF Connect SDK v1.5.0`_ for the list of issues valid for this release.
```
Not sure if this is a temporary thing to get the documentation building, but in case it isn't:

```suggestion
See `known issues for nRF Connect SDK v1.6.0`_ for the list of issues valid for this release.
```

b-gent replied:
Yes, that will be fixed

greg-fer commented:
Summoning @Damian-Nordic for Matter-related release notes.

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+
+The |NCS| is where you begin building low-power wireless applications with Nordic Semiconductor nRF52, nRF53, and nRF91 Series devices.
+nRF53 Series devices are now supported for production.
+Wireless protocol stacks and libraries may indicate support for development or support for production for different series of devices based on verification and certification status. See the release notes and documentation for those protocols and libraries for further information.
```
- Move the second sentence to a new line.
- If I recall, there were talks about what "production" and "development" actually mean. Not sure there was any consensus, but shouldn't we explain here briefly what these terms mean?

b-gent replied:
Production and development terminology is under discussion but we do not change anything for this release.

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+* Added production support for multiple advertising sets to SoftDevice Controller.
+* Added features available in Trusted Firmware-M (TF-M) which is supported for development with nRF5340 and nRF9160.
+* Integrated Apple Find My supported for production.
+* Integrated Pelion cloud client supported for development for LTE-M/NB-IoT and Thread.
```
```suggestion
* Added development support for the Pelion Device Management library for LTE-M/NB-IoT and Thread.
```
@pdunaj to confirm the suggestion.

b-gent replied:
leaving this as is

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+* Added production support for nRF5340 for Bluetooth LE in single protocol configuration and in multiprotocol operation with Thread for Bluetooth Peripheral role.
+* Added production support for multiple advertising sets to SoftDevice Controller.
+* Added features available in Trusted Firmware-M (TF-M) which is supported for development with nRF5340 and nRF9160.
+* Integrated Apple Find My supported for production.
```
```suggestion
* Integrated Apple Find My, supported for production.
```
or
```suggestion
* Added production support for Apple Find My.
```

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+* Added features available in Trusted Firmware-M (TF-M) which is supported for development with nRF5340 and nRF9160.
+* Integrated Apple Find My supported for production.
+* Integrated Pelion cloud client supported for development for LTE-M/NB-IoT and Thread.
+* Edge Impulse Machine Learning has been further integrated with an Application example available for Thingy:52.
```
```suggestion
* Added the :ref:`nrf_machine_learning_app` application for Thingy:52, which further integrates Edge Impulse Machine Learning with the |NCS|.
```
@pdunaj to confirm suggestion.

b-gent replied:
changed

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,471 @@
+* Integrated Apple Find My supported for production.
+* Integrated Pelion cloud client supported for development for LTE-M/NB-IoT and Thread.
+* Edge Impulse Machine Learning has been further integrated with an Application example available for Thingy:52.
+* Added complete support for Thread 1.2 mandatory features for routers and end devices supported for development for the nRF52 family.
```
```suggestion
* Added complete support for Thread 1.2 mandatory features for routers and end devices supported for development for the nRF52 Series family.
```

What does `supported for development` refer to here, @edmont ? "complete support" as in previous bullet points or "routers and end devices", as the context would suggest?

edmont replied:
Maybe like this is better explained:
> Added support for all Thread 1.2 mandatory features...

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+* Integrated Pelion cloud client supported for development for LTE-M/NB-IoT and Thread.
+* Edge Impulse Machine Learning has been further integrated with an Application example available for Thingy:52.
+* Added complete support for Thread 1.2 mandatory features for routers and end devices supported for development for the nRF52 family.
+* Integrated more complete support for Matter (Project CHIP) which is supported for development for nRF52840 and nRF5340.
```
```suggestion
* Matter (formerly Project CHIP) is now supported for development for nRF52840 and nRF5340.
```

b-gent replied:
I think it was already supported for development for nRF52840 and nRF5340.

b-gent replied:
I will leave as is for now, PMT will be looking at this anyway

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+* Integrated more complete support for Matter (Project CHIP) which is supported for development for nRF52840 and nRF5340.
+* Integrated Apple HomeKit ADK v5.3 and added development support for nRF52832 and nRF52833 for Bluetooth LE HomeKit accessories.
+* Zigbee Network Co-Processor (NCP) design is now supported for production for nRF52833 and nRF52840.
+* The application layer part of ZBOSS Zigbee stack is available in source code now.
```
```suggestion
* Added the application layer part of the ZBOSS Zigbee stack to source code.
```

b-gent replied:
I think this changes the meaning a bit, how I understand the original sentence is that the part is now available in source code *format*, not that it is added to source code.

greg-fer replied:
Oh, right.

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+nRF9160
+=======
+
+* Added:
```
Should we add a mention of the nRF Pelion Client application in this section, as it supports the nRF9160 DK? @pdunaj?

pdunaj replied:
I have added Pelion it to "Common" section as it is both for nrf91 and nrf5x. To be honest I don't get it how this nrf5x vs nrf91 helps as most things except modem and BLE radio are common and these two could be components as any other.

b-gent replied:
leaving as is

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+  * Coordinated Sampled Listening (CSL)
+
+  Link Metrics and CSL are supported for nRF52 Series devices.
+
```
Blank line not needed.

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+  * Link Metrics
+  * Coordinated Sampled Listening (CSL)
+
+  Link Metrics and CSL are supported for nRF52 Series devices.
```
```suggestion
 Link Metrics and CSL are supported for the nRF52 Series devices.
```

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+  Link Metrics and CSL are supported for nRF52 Series devices.
+
+* NCP sample renamed to Co-processor, with added support for RCP architecture.
```
```suggestion
* NCP sample renamed to :ref:`Co-processor <ot_coprocessor_sample>`, with added support for the :ref:`thread_architectures_designs_cp_rcp` architecture.
```

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+Zigbee
+------
+
+* Added the v0.9.5 version of the `ZBOSS NCP Host`_ package that includes a simple gateway application.
```
No need to repeat `version` here.

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+* Updated:
+
+  * Renamed occurrences of Project CHIP to Matter.
```
Add line that references the submodule section:
```
  * Updated the Matter fork in the |NCS| to the revision mentioned in the `Matter (Project CHIP)`_ submodule section below.
```

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+      * :ref:`ug_bootloader_external_flash`
+      * :ref:`ug_bootloader_config`
+      * :ref:`ug_fw_update`
+   * :ref:`app_bootloaders` - Bootloader and firmware update guide header page
```
Add blank line before this line.

umapraseeda commented on doc/nrf/links.txt:
```diff
@@ -38,6 +38,7 @@
 .. _`Changelog for nrfx 2.2.0`: https://github.com/NordicSemiconductor/nrfx/blob/master/CHANGELOG.md#user-content-220---2020-04-28
 .. _`Changelog for nrfx 2.3.0`: https://github.com/NordicSemiconductor/nrfx/blob/master/CHANGELOG.md#user-content-230---2020-08-19
 .. _`Changelog for nrfx 2.4.0`: https://github.com/NordicSemiconductor/nrfx/blob/master/CHANGELOG.md#user-content-240---2020-11-13
+.. _`Changelog for nrfx 2.5.0`: https://github.com/NordicSemiconductor/nrfx/blob/master/CHANGELOG.md
```
```suggestion
.. _`Changelog for nrfx 2.5.0`: https://github.com/NordicSemiconductor/nrfx/blob/master/CHANGELOG.md#250---2021-05-05
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+.. _ncs_release_notes_160:
+
+Changes in |NCS| v1.6.0
+#######################
```
Make sure to clean the changelog since the contents have been moved here. Also, the link to the changelog should be removed from doc/nrf/release_notes.rst
```suggestion
|NCS| v1.6.0 Release Notes
##########################
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+* Added production support for nRF5340 for Bluetooth LE in single protocol configuration and in multiprotocol operation with Thread for Bluetooth Peripheral role.
+* Added production support for multiple advertising sets to SoftDevice Controller.
+* Added features available in Trusted Firmware-M (TF-M) which is supported for development with nRF5340 and nRF9160.
```
```suggestion
* Added features available in Trusted Firmware-M (TF-M), which is supported for development with nRF5340 and nRF9160.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+* Added:
+
+  * :ref:`lib_modem_jwt` library, which provides an API to obtain a JSON Web Token (JWT) from the modem.  Functionality requires modem firmware v1.3.0 or higher.
```
```suggestion
  * :ref:`lib_modem_jwt` library:
  
     * The library provides an API to obtain a JSON Web Token (JWT) from the modem.
     * Functionality requires modem firmware v1.3.0 or higher.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+    * The library provides an API to get an attestation token from the modem.
+    * Functionality requires modem firmware v1.3.0 or higher.
+
+  * :ref:`mod_memfault` module and integration on |NCS| for nRF9160-based devices.
```
```suggestion
  * :ref:`mod_memfault` module and integration on |NCS| for nRF9160-based devices:
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+    * Integration of Memfault SDK into |NCS|.
+
+  * :ref:`memfault_sample` sample implementing :ref:`mod_memfault` module and showing how to use Memfault SDK functionality in an application.
```
```suggestion
  * :ref:`memfault_sample` sample:
  
    * Implements :ref:`mod_memfault` module and shows how to use Memfault SDK functionality in an application.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+  * :ref:`memfault_sample` sample implementing :ref:`mod_memfault` module and showing how to use Memfault SDK functionality in an application.
+
+  * :ref:`modem_shell_application` enables to test various connectivity features such as link handling, TCP/IP connections, data throughput (curl and iPerf3), SMS, GNSS, FOTA and PPP.
```
```suggestion
  * :ref:`modem_shell_application` sample:
  
    * Enables testing of various connectivity features such as link handling, TCP/IP connections, data throughput (curl and iPerf3), SMS, GNSS, FOTA and PPP.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+    * The library adds P-GPS (Predicted GPS) support to the :ref:`lib_nrf_cloud` library.
+
+  * :ref:`sms_sample` demonstrates how you can send and receive SMS messages with your nRF9160-based device using the :ref:`sms_readme`.
```
```suggestion
  * :ref:`sms_sample` sample:
  
    * Demonstrates how you can send and receive SMS messages with your nRF9160-based device using the :ref:`sms_readme` library.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+  * :ref:`sms_sample` demonstrates how you can send and receive SMS messages with your nRF9160-based device using the :ref:`sms_readme`.
+
+  * :ref:`pdn_sample` demonstrates how to create and configure a Packet Data Protocol (PDP) context, activate a Packet Data Network connection, and receive events on its state and connectivity using the :ref:`pdn_readme` library.
```
```suggestion
  * :ref:`pdn_sample` sample:
  
    * Demonstrates how to create and configure a Packet Data Protocol (PDP) context, activate a Packet Data Network connection, and receive events on its state and connectivity using the :ref:`pdn_readme` library.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+  * :ref:`pdn_sample` demonstrates how to create and configure a Packet Data Protocol (PDP) context, activate a Packet Data Network connection, and receive events on its state and connectivity using the :ref:`pdn_readme` library.
+
+  * :ref:`pdn_readme` library can be used to manage Packet Data Protocol (PDP) contexts and PDN connections.
```
```suggestion
  * :ref:`pdn_readme` library:
  
    * Manages Packet Data Protocol (PDP) contexts and PDN connections.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+  * :ref:`lib_nrf_cloud` library:
+
+    * Added cellular positioning support to :ref:`lib_nrf_cloud_cell_pos`.
+    * Added Kconfig option :option:`CONFIG_NRF_CLOUD_CELL_POS` to obtain cell-based location from nRF Cloud instead of using the modem's GPS.
```
This could be moved to 'Added' section: 

```suggestion
 *  :ref:`lib_nrf_cloud_cell_pos` library: 
    * Added cellular positioning support to :ref:`lib_nrf_cloud` library.
    * Added Kconfig option :option:`CONFIG_NRF_CLOUD_CELL_POS` to obtain cell-based location from nRF Cloud instead of using the modem's GPS.
```

b-gent replied:
I will leave as is. The Added section contains only completely new libs and samples.

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+    * Added cellular positioning support to :ref:`lib_nrf_cloud_cell_pos`.
+    * Added Kconfig option :option:`CONFIG_NRF_CLOUD_CELL_POS` to obtain cell-based location from nRF Cloud instead of using the modem's GPS.
+    * Added function :c:func:`nrf_cloud_modem_fota_completed` which is to be called by the application after it re-initializes the modem (instead of rebooting) after a modem FOTA update.
```
```suggestion
    * Added function :c:func:`nrf_cloud_modem_fota_completed`, which is to be called by the application after it re-initializes the modem (instead of rebooting) after a modem FOTA update.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+    * Added support for %XT3412 AT command notifications, which allows the application to get prewarnings before Tracking Area Updates.
+    * Added support for neighbor cell measurements.
+    * Added support for %XMODEMSLEEP AT command notifications which allows the application to get notifications related to modem sleep.
```
```suggestion
    * Added support for %XMODEMSLEEP AT command notifications, which allows the application to get notifications related to modem sleep.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+    * Added data mode to the MQTT Publish service to support JSON-type payload.
+    * Added SMS support, to send/receive SMS in plain text.
+
+  * :ref:`at_cmd_parser_readme`:
```
```suggestion
  * :ref:`at_cmd_parser_readme` library:
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+    * Fixed an issue where SPM and the application could have incompatible FPU configurations, resulting in a HardFault.
+      Now, the application is free to use FPU regardless of SPM configuration.
+
+  * :ref:`gps_api`:
```
```suggestion
  * :ref:`gps_api` interface:
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+  * :ref:`aws_fota_sample` sample:
+
+    * Removed nRF Connect for Cloud support code, because ``fota_v1`` is no longer supported in nRF Connect for Cloud.
```
```suggestion
    * Removed nRF Cloud support code, because ``fota_v1`` is no longer supported in nRF Cloud.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+    * Updated internal state handling and fault tolerance.
+
+  * :ref:`sms_readme` updated to enable sending of sms and decoding of received SMS payload.
```
```suggestion
  * :ref:`sms_readme` library:
  
    * Updated to enable sending of SMS and decoding of received SMS payload.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+  * :ref:`sms_readme` updated to enable sending of sms and decoding of received SMS payload.
+
+  * :ref:`liblwm2m_carrier_readme` library - Updated to v0.20.1. See :ref:`liblwm2m_carrier_changelog` for details.
```
```suggestion
  * :ref:`liblwm2m_carrier_readme` library:
  
     * Updated to v0.20.1. See :ref:`liblwm2m_carrier_changelog` for details.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+  * :ref:`liblwm2m_carrier_readme` library - Updated to v0.20.1. See :ref:`liblwm2m_carrier_changelog` for details.
+
+  * :ref:`lib_lwm2m_client_utils` created from common parts of :ref:`lwm2m_client`. This module can be reused to add common objects to LwM2M applications.
```
Could be moved to 'Added' section: 
```suggestion
  * :ref:`lib_lwm2m_client_utils` library:
  
    * Created from common parts of :ref:`lwm2m_client` sample.
    * This module can be reused to add common objects to LwM2M applications.
```

b-gent replied:
moved

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+Modem library
++++++++++++++
+
+* Updated Modem library to version 1.2.1.
```
```suggestion
* Updated :ref:`nrf_modem` to version 1.2.1.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+* Updated Modem library to version 1.2.1.
+  See the :ref:`nrfxlib:nrf_modem_changelog` for detailed information.
+* Added a new function based GNSS API with support for new GNSS features in MFW 1.3.0.
```
```suggestion
* Added a new function-based GNSS API with support for new GNSS features in modem firmware v1.3.0.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,488 @@
+
+* Added support for indicating serial recovery through LED.
+* Made the debounce delay of the serial detect pin state configurable.
+* Added support for mbed TLS ECDSA for signatures.
```
```suggestion
* Added support for Mbed TLS ECDSA for signatures.
```

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,564 @@
+
+* :ref:`ug_zigbee` - The following sections were added or changed considerably:
+
+   * :ref:`ug_zigbee_architectures`
```
This section barely changed, and this mention can be removed.

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,564 @@
+* :ref:`ug_zigbee` - The following sections were added or changed considerably:
+
+   * :ref:`ug_zigbee_architectures`
+   * :ref:`zigbee_memory`
```
```suggestion
   * :ref:`zigbee_memory` - Updated the memory values for the latest release.
```

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,564 @@
+
+   * :ref:`ug_zigbee_architectures`
+   * :ref:`zigbee_memory`
+   * :ref:`ug_zigbee_other_ecosystems`
```
```suggestion
   * :ref:`ug_zigbee_other_ecosystems` - New page.
```

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,564 @@
+   * :ref:`ug_zigbee_architectures`
+   * :ref:`zigbee_memory`
+   * :ref:`ug_zigbee_other_ecosystems`
+   * :ref:`ug_zigbee_tools`
```
```suggestion
   * :ref:`ug_zigbee_tools` - Updated with new content and structure.
```

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,564 @@
+* :ref:`ug_multi_image` - Added more information regarding child image usage, configuration options, and image-specific variables.
+* :ref:`partition_manager` - Added section about partition reports.
+* :ref:`ug_tfm` - Added references to new crypto samples that utilize TF-M and information about the TF-M minimal build.
+* :ref:``ug_thread` - The following sections were added or changed considerably:
```
Double tick after `:ref:`: ``

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,564 @@
+   * :ref:`asset_tracker` - Added sections on using nRF Cloud A-GPS or P-GPS, and on using nRF Cloud FOTA.
+   * :ref:`asset_tracker_v2` - Added a table showing cloud services and the corresponding cloud-side instances.
+   * :ref:`serial_lte_modem` - Added links to AT command reference guides.
+   * :ref: `agps_sample` - Added a section on using nRF Cloud A-GPS or P-GPS.
```
Space after :ref:

greg-fer reviewed (commented):
Some comments for the Documentation section.

shantha-14 commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,564 @@
+Supported modem firmware
+************************
+
+This version of the |NCS| has been tested with the following modem firmware for cellular IoT applications:
```
We agreed we will remove the MFW from release notes but instead point to the compatibility matrix instead addd a link here

b-gent commented:
@pdunaj  could you respond in the comments where you are tagged?

b-gent commented:
@ioannisg  @tejlmand  Can you or someone from Vestavind confirm that the following are correct:
- commit hash of MCUboot -> 2fce9769b1
- commit hash of Mcumgr -> 74e77ad08 ( this is for sure **not** correct as this has not been updated since last version)
- commit hash of Zephyr -> ``730acbd6ed`` (``v2.6.0-rc1``)

We also need a very short summary of changes coming from Mcumgr , as no one has added this. Something like this:

http://developer.nordicsemi.com/nRF_Connect_SDK/doc/1.5.0/nrf/releases/release-notes-1.5.0.html#mcumgr

b-gent commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,587 @@
+
+  * Support for serialization of the :ref:`zephyr:bt_gap` and the :ref:`zephyr:bluetooth_connection_mgmt` API.
+  * :ref:`ble_rpc_host` sample that enables support for serialization of the :ref:`zephyr:bt_gap` and the :ref:`zephyr:bluetooth_connection_mgmt`.
+  * Samples demonstrating the Direction Finding feature (available for the nRF52833 DK):
```
@pdunaj  adding this as discussed, please verify

pdunaj replied:
I am ok, @ppryga any else to be added?

ppryga replied:
```suggestion
* Samples demonstrating the direction finding feature in connectionless mode (available for the nRF52833 DK):
```
@mia-ko has checked BT core spec and the direction finding feature isn't written there with capitals. Also I suggest to state it explicitly that there is support for direction finding based on periodic advertising (connectionless), to avoid confusion for customers. Besides that I'm ok with it.

b-gent replied:
done, thanks

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,587 @@
+   * :ref:`asset_tracker_v2` - Added a table showing cloud services and the corresponding cloud-side instances.
+   * :ref:`serial_lte_modem` - Added links to AT command reference guides.
+   * :ref:`agps_sample` - Added a section on using nRF Cloud A-GPS or P-GPS.
+   * :ref:`aws_fota_sample` - Changes in the Sample configuration section.
```
```suggestion
   * :ref:`aws_fota_sample` - Changes in the sample configuration section.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,587 @@
+* Added a new documentation set for :doc:`nrfx <nrfx:index>` - a standalone set of drivers for peripherals.
+* Project CHIP has been renamed to Matter in all occurrences throughout the documentation.
+* :ref:`glossary` - Added a new, comprehensive glossary that explains many of the terms used when working with the |NCS|.
+* :ref:`doc_build` - Added a new section on dealing with warnings during doc builds.
```
```suggestion
* :ref:`doc_build` - Added a new section on dealing with warnings during documentation builds.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,587 @@
+  TF-M automatically generates and stores random hardware unique keys (if not present), using the :ref:`lib_hw_unique_key` library.
+* TFM_MINIMAL: Added a size-optimized configuration of TF-M which provides a minimal set of features:
+
+  * This configuration requires 32 kB of flash and provides random number generation, SHA-256, and the platform memory read service, which is analogous to the feature set of :ref:`secure_partition_manager`.
```
```suggestion
  * This configuration requires 32 KB of flash and provides random number generation, SHA-256, and the platform memory read service, which is analogous to the feature set of :ref:`secure_partition_manager`.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,587 @@
+  * Replaced the linked list of scene entries in the model contexts, with a lookup in ROM-allocated scene entries.
+  * Updated so the transition pointer can be NULL, if no transition time parameters are provided in APIs.
+  * Renamed Kconfig option ``CONFIG_BT_MESH_LIGHT_CTRL_STORE_TIMEOUT`` to :option:`CONFIG_BT_MESH_MODEL_SRV_STORE_TIMEOUT`, and default value is set to 0.
+  * Updated the :ref:`bt_mesh_light_ctrl_srv_readme` with a timer, allowing it to resume operation after a certain delay.
```
```suggestion
  * Updated the :ref:`bt_mesh_light_ctrl_srv_readme` model with a timer, allowing it to resume operation after a certain delay.
```

umapraseeda commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,587 @@
+
+* Updated:
+
+  * Updated the :ref:`bt_mesh_light_xyl_srv_readme` to publish its state values that were loaded from flash after powering up.
```
```suggestion
  * Updated the :ref:`bt_mesh_light_xyl_srv_readme` model to publish its state values that were loaded from flash after powering up.
```

de-nordic commented:
> 
> 
> @ioannisg @tejlmand Can you or someone from Vestavind confirm that the following are correct:
> 
>     * commit hash of MCUboot -> 2fce9769b1
> 
>     * commit hash of Mcumgr -> 74e77ad08 ( this has not been updated since last version, have we updated this component since then?)
> 
>     * commit hash of Zephyr -> `730acbd6ed` (`v2.6.0-rc1`)
> 
> 
> We also need a very short summary of changes coming from Mcumgr , as no one has added this. Something like this:
> 
> http://developer.nordicsemi.com/nRF_Connect_SDK/doc/1.5.0/nrf/releases/release-notes-1.5.0.html#mcumgr
> 
> (this is assuming that we have updated Mcumgr)
> 
> Edit: Seems like @de-nordic is the person to answer about Mcumgr

The mcumgr hash is directly taken from Zephyr we update to. The  current top on upstream and nrf is 5c5055f5a7565f8152d75fcecf07262928b4d56e, so basically the release notes for 2.6 can be reused:
```
  * MCUmgr

    * Fixed an issue with the file system management failing to
      open files due to missing initializations of `fs_file_t` structures.
    * Fixed an issue where multiple SMP commands sent one after the other would
      corrupt CBOR payload.
    * Fixed problem where mcumgr over shell would stall and wait for
      retransmissions of frames.
```
Note only that the first issue is actually fixed in mcumgr library, the others are fixed in Zephyr but affect mcumgr.

b-gent commented:
Changed the mcumgr sha to 5c5055f5a and added the list of changes provided by @de-nordic 

Still waiting for confirmation on Zephyr and MCUboot @ioannisg  :)

b-gent commented:
Added Workqueue migration guide in the Zephyr section.

b-gent commented:
HTML output is here -> http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-4966/nrf/releases/release-notes-1.6.0.html

b-gent commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,651 @@
+Zigbee
+------
+
+In this release, Zigbee is supported for development and should not be used for production.
```
@gkafel please confirm this is correct

gkafel replied:
PMT wanted indication what is the Zigbee release which could be used for production (namely: NCS 1.5.1)

b-gent replied:
it can be found in the next line

b-gent commented:
Removed from Highlights:
"Zigbee Network Co-Processor (NCP) design is now supported for production for nRF52833 and nRF52840"
, as requested by PMT

jdstapes commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,650 @@
+* :ref:`nrf_security_readme` - Added section on Building with TF-M.
+* Added :ref:`zboss_certification`.
+
+Known issues
+************
+
+See `known issues for nRF Connect SDK v1.6.0`_ for the list of issues valid for this release.
```
Propose: move known issues above "change log"

```suggestion
Known issues
************

See `known issues for nRF Connect SDK v1.6.0`_ for the list of issues valid for this release.

Important known issues
----------------------------
These known issues are potential regressions or high impact issues.
SoftDevice Controller DRGN-15852 and DRGN-15852: Affecting scanning on the nRF53 Series.
HomeKit KRKNWK-10611: DFU will fail when using external flash memory for update image storage.
```

b-gent replied:
This is done now.

thomasstenersen commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,658 @@
+
+These known issues are potential regressions or high impact issues:
+
+* :ref:`SoftDevice Controller DRGN-15852 and DRGN-15852 <drgn_15852>`: Affects scanning on the nRF53 Series.
```
It says *DRGN-15852 and DRGN-15852*, which is the same issue number (and root cause). Let's remove the duplicate:

```suggestion
* :ref:`SoftDevice Controller DRGN-15852 <drgn_15852>`: Affects scanning on the nRF53 Series.
```

b-gent replied:
fixed

greg-fer commented on doc/nrf/releases/release-notes-1.6.0.rst:
```diff
@@ -0,0 +1,658 @@
+Edge Impulse
+------------
+
+* Added the :ref:`nrf_machine_learning_app` application that integrates the Edge Impulse wrapper library with sensor sampling.
+* Sample reference to a public pre-trained model.
```
I was sure I commented on this line, but looks like I haven't:
```suggestion
Added the :ref:`nrf_machine_learning_app` application that integrates the Edge Impulse wrapper library and the Edge Impulse data forwarder with sensor sampling.
```
Discussed and approved with @MarekPieta

b-gent replied:
Sorry @greg-fer  @MarekPieta  but it's too late for this change to go in
